File: lc/0001-two-sum/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int i = 0;
        int[] list = new int[2];
        while (i < nums.length) {
            // if (target - nums[i] )
            for (int j = i + 1; j < nums.length; j++) {
                if (target - nums[i] == nums[j]) {
                    // list.add(i);
                    // list.add(j);
                    list[0] = i;
                    list[1] = j;
                }
            }
            i++;
        }
        return list;
    }
}

File: lc/0001-two-sum/solution.py
from collections import defaultdict
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        visited = defaultdict(list)
        for idx, num in enumerate(nums):
            visited[num] += [idx]
        
        for i, num in enumerate(nums):
            if num in visited:
                if len(visited[num]) == 1:
                    del visited[num]
                else:
                    visited[num].pop(0)
            complement = target - num
            if complement in visited:
                return [i, visited[complement][-1]]
        
        return [0, 1]
